students/alex_skrn/lesson05/mailroom.py

```diff
@@ -0,0 +1,244 @@
+        write_file(cwd, name, text)
+
+    print("\nAll letters saved in {}\n".format(cwd))
+
+
+def write_select_dir():
+    """Write all emails to a dir selected by the user."""
+    root = tk.Tk()
+    root.withdraw()
+
+    # Get the target directory from the user.
+    target_dir = filedialog.askdirectory()
+    if not target_dir:  # If the user hits cancel.
+        return
+    for name in donors:
+        text = get_email(name, get_last_donation(name))
+        write_file(target_dir, name, text)
+
+    print("\nAll letters saved in {}\n".format(target_dir))
+
+
+# MANAGING MENUS
+def quit():
+    """Provide an exit option for menus."""
+    return "exit menu"
+
+
+def send_all_menu():
+    """Initiate the send-all-letters sub-sub-menu."""
+    menu_selection(write_file_prompt, write_file_dispatch)
+
+
+def send_thank_you_interaction():
+    """Initiate the send-thank-you sub-menu."""
+    menu_selection(send_thanks_prompt, send_thanks_dispatch)
+
+
+def menu_selection(prompt, dispatch_dict):
+    """Provide a template for using dispatch dicts to switch through menus."""
+    while True:
+        response = input(prompt)
+        try:
+            if dispatch_dict[response]() == "exit menu":
+                break
+        except KeyError:
+            print("\nInvalid choice. Try again")
+
+
+if __name__ == "__main__":
+    #  Write to files
+    write_file_prompt = ("\nSend to everyone sub-menu\n"
+                         "\n1 - Write to current working directory\n"
+                         "2 - Choose a directory to write\n"
+                         "3 - Quit\n"
+                         ">> "
+                         )
+    write_file_dispatch = {"1": write_cwd,
+                           "2": write_select_dir,
+                           "3": quit,
+                           }
+    # Print on screen
+    send_thanks_dispatch = {"1": print_donor_names,
+                            "2": new_donor_interaction,
+                            "3": existing_donor_interaction,
+                            "4": quit,
+                            }
+    send_thanks_prompt = ("\nSend-Thank-You Sub-Menu\n"
+                          "\n1 - See the list of donors\n"
+                          "2 - Add a new donor and a donation amount\n"
+                          "3 - Choose an existing donor\n"
+                          "4 - Quit\n"
+                          ">> "
+                          )
+    # Main menu
+    main_dispatch = {"1": send_thank_you_interaction,
+                     "2": create_report,
+                     "3": send_all_menu,
+                     "4": quit,
+                     }
+    main_prompt = ("\nMain Menu\n"
+                   "\n1 - Send a Thank You\n"
+                   "2 - Create a Report\n"
+                   "3 - Send letters to everyone\n"
+                   "4 - Quit\n"
+                   ">> "
+                   )
+    menu_selection(main_prompt, main_dispatch)
```

Comment:
All of these above would be better served as functions than a huge list of menus under the `if __name__ == "__main__":` prompt

Reply:
Thank you, I will do this in the next iteration - in lesson 6 which deals with testing.